test/multitenancy/loginMethods.test.js

```diff
@@ -0,0 +1,314 @@
+/* Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+const { printPath, setupST, startSTWithMultitenancy, killAllST, cleanST } = require("../utils");
+let assert = require("assert");
+const express = require("express");
+const request = require("supertest");
+let { Querier } = require("../../lib/build/querier");
+let { ProcessState } = require("../../lib/build/processState");
+let SuperTokens = require("../../");
+let Multitenancy = require("../../recipe/multitenancy");
+let EmailPassword = require("../../recipe/emailpassword");
+let ThirdPartyEmaillPassword = require("../../recipe/thirdpartyemailpassword");
+let Passwordless = require("../../recipe/passwordless");
+let MultifactorAuth = require("../../recipe/multifactorauth");
+let Session = require("../../recipe/session");
+let { middleware, errorHandler } = require("../../framework/express");
+
+describe(`loginMethods: ${printPath("[test/multitenancy/loginMethods.test.js]")}`, function () {
+    beforeEach(async function () {
+        await killAllST();
+        await setupST();
+        ProcessState.getInstance().reset();
+    });
+
+    after(async function () {
+        await killAllST();
+        await cleanST();
+    });
+
+    it("test without mfa - case 1", async function () {
```

Comment:
as a minor nitpick, the general recommendation is to have titles like (notice the should and the phrasing:
```suggestion
    it("should return firstFactors based on enabled recipes if no other configuration is available", async function () {
```

Reply:
you can add describes in here like instead of adding a prefix to test titles (if you think it's necessary):
`describe("without MFA initialised", () => {`

Reply:
Done